lib/collection.dart

```diff
@@ -2,6 +2,20 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+// TODO(lrn): Move the classes to this package.
+export "dart:collection" show
+    DoubleLinkedQueue,
+    DoubleLinkedQueueEntry,
+    HasNextIterator,
+    LinkedList,
+    LinkedListEntry,
+    MapView, // TODO(lrn): Drop this class, it is equivalent to DelegatingMap.
+    SplayTreeMap,
+    SplayTreeSet,
+    UnmodifiableListView,
+    UnmodifiableMapBase,
+    UnmodifiableMapView;
```

Comment:
The unmodifiable classes are widely used. I think we should consider leaving them in `dart:collection`.

Reply:
Being widly used isn't by itself enough to warrant a place in the platform libraries. We have `package:collection` (and friends) to hold generally useful code that doesn't *need* to be in the platform libraries.

Reply:
These classes are [used occasionally](https://github.com/dart-lang/sdk/blob/cc5cb89e0b4ec788ad3b37d577f49ca7fa39b886/sdk/lib/io/platform_impl.dart#L110) in the SDK, but I'm also skeptical of the principle of stripping absolutely everything from the core libraries that's not strictly necessary for representing the transitive closure of the language spec and `dart:io`'s APIs or whatever.

The unmodifiable classes' APIs move in lock-step with the collection APIs, so they don't get any added flexibility from being in a package. They're not only widely-used, they're a fundamental part of exposing an API surface with strong encapsulation guarantees. I don't see the benefit of removing them from the core libraries, whereas requiring many users to depend on `collection` is a real cost.